uaa/src/test/java/org/cloudfoundry/identity/uaa/integration/pageObjects/LogoutDoEndpoint.java

```diff
@@ -0,0 +1,26 @@
+package org.cloudfoundry.identity.uaa.integration.pageObjects;
+
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+
+import org.openqa.selenium.WebDriver;
+
+public class LogoutDoEndpoint extends Page {
+    static final protected String urlPath = "/logout.do";
+
+
+    public LogoutDoEndpoint(WebDriver driver) {
+        super(driver);
+    }
+
+    static public LoginPage logout_goToLoginPage(WebDriver driver, String baseUrl, String redirectUrl, String clientId) {
+        driver.get(
+                baseUrl
+                + urlPath
+                + "?redirect=" + URLEncoder.encode(redirectUrl, StandardCharsets.UTF_8)
+                + "&client_id=" + clientId
+        );
+        return new LoginPage(driver);
+    }
```

Comment:
I guess I just don't understand the pattern here.

Reply:
What part of it is confusing? The method name? The code in the body?

Reply:
This one appears to make more sense, now that you explained.

So this method is to simulate clicking the "logout" button, which redirects you to the login page.

But the code doesn't seem to by logging anybody out.

Reply:
It's hitting logout.do. I extracted a method in hope of making that more clear.

It's not directly clicking the logout button, but taking the same short of shortcut that the tests were using before the refactor, by hitting the endpoint directly.